File: android2_lesson04/src/java/java/com/example/ama/android2_lesson04/ui/viewer/PVFragmentIntentService.java
package com.example.ama.android2_lesson04.ui.viewer;

import android.content.Intent;

import com.example.ama.android2_lesson04.background.service.LoadPictureIntentService;
import com.example.ama.android2_lesson04.background.service.ServiceConstants;
import com.example.ama.android2_lesson04.ui.viewer.base.BaseReceiverFragment;
import com.example.ama.android2_lesson04.utils.ResourcesUtils;

public class PVFragmentIntentService extends BaseReceiverFragment {

    public static PVFragmentIntentService newInstance() {
        return new PVFragmentIntentService();
    }

    @Override
    protected void onLoadImagesClick() {
        startIntentService();
    }

    private void startIntentService() {
        if (getActivity() != null) {
            Intent intent = new Intent(getActivity(), LoadPictureIntentService.class);
            intent.putExtra(ServiceConstants.EXTRA_KEY, ResourcesUtils.getUrlsArray());
            getActivity().startService(intent);
        }
    }
}

File: android2_lesson04/src/kotlin/java/com/example/ama/android2_lesson04/ui/viewer/PVFragmentIntentService.kt
package com.example.ama.android2_lesson04.ui.viewer

import android.content.Intent
import com.example.ama.android2_lesson04.R
import com.example.ama.android2_lesson04.ServiceTestApp
import com.example.ama.android2_lesson04.background.service.LoadPictureIntentService
import com.example.ama.android2_lesson04.ui.viewer.base.BaseReceiverFragment
import com.example.ama.android2_lesson04.ui.viewer.base.EXTRA_KEY

class PVFragmentIntentService : BaseReceiverFragment() {

    companion object {
        fun newInstance() = PVFragmentIntentService()
    }

    override fun onLoadImagesClick() {
        startIntentService()
    }

    private fun startIntentService() {
        val intent = Intent(ServiceTestApp.instance, LoadPictureIntentService::class.java)
        intent.putExtra(EXTRA_KEY, resources.getStringArray(R.array.image_urls))
        activity?.startService(intent)
    }
}